src/cmd/ksh93/sh/io.c

```diff
@@ -1990,14 +1991,20 @@ static int io_prompt(Shell_t *shp, Sfio_t *iop, int flag) {
 #endif /* TIOCLBIC */
             cp = sh_mactry(shp, nv_getval(sh_scoped(shp, PS1NOD)));
             shp->exitval = 0;
-            for (; (c = *cp); cp++) {
-                if (c == HIST_CHAR) {
+            for (escape_index = 0; (c = *cp); cp++) {
```

Comment:
Might as well put the declaration inside the `for` statement since the var isn't used outside the loop.